src/Docker.DotNet/Endpoints/StreamUtil.cs

```diff
@@ -29,16 +28,10 @@ internal static async Task MonitorStreamAsync(Task<Stream> streamTask, DockerCli
 
         internal static async Task MonitorStreamForMessagesAsync<T>(Task<Stream> streamTask, DockerClient client, CancellationToken cancellationToken, IProgress<T> progress)
         {
-            var tcs = new TaskCompletionSource<bool>();
-
             using (var stream = await streamTask)
-            using (var reader = new StreamReader(stream, new UTF8Encoding(false)))
-            using (var jsonReader = new JsonTextReader(reader) { SupportMultipleContent = true })
-            using (cancellationToken.Register(() => tcs.TrySetCanceled(cancellationToken)))
             {
```

Comment:
Verify this task can be cancelled by running the test `CreateContainerAsync_TimeoutExpires_Fails`.